File: src/scanner/ignore.ts
/** Directory names skipped during repo walk and filtered from surfaces/routing. */
export const IGNORE_DIR_NAMES = new Set([
  ".git",
  "node_modules",
  "dist",
  "build",
  "out",
  ".next",
  "coverage",
  "vendor",
  ".turbo",
  ".pnpm-store",
  ".yarn",
  ".venv",
  "venv",
  "env",
  "__pycache__",
  ".pytest_cache",
  "site-packages",
  ".cursor",
  ".mypy_cache",
  ".ruff_cache",
  ".tox",
  ".cache",
]);

export const IGNORE_RELATIVE_PREFIXES = [".swarm/cache", ".swarm/reports"];

export function normalizeRelativePath(relPath: string): string {
  return relPath.replace(/\\/g, "/");
}

export function shouldIgnoreDirName(name: string): boolean {
  return IGNORE_DIR_NAMES.has(name);
}

/** True when any path segment is a dependency/runtime artifact directory. */
export function shouldIgnoreScannedPath(relPath: string): boolean {
  const normalized = normalizeRelativePath(relPath);
  if (
    IGNORE_RELATIVE_PREFIXES.some(
      (prefix) => normalized === prefix || normalized.startsWith(`${prefix}/`),
    )
  ) {
    return true;
  }

  for (const segment of normalized.split("/")) {
    if (IGNORE_DIR_NAMES.has(segment)) {
      return true;
    }
  }

  return false;
}

File: agent_runtime/cyber_swarm/evidence/ignore.py
"""Path ignore rules aligned with TypeScript scanner ignore.ts."""

from __future__ import annotations

IGNORE_DIR_NAMES = frozenset(
    {
        ".git",
        "node_modules",
        "dist",
        "build",
        "out",
        ".next",
        "coverage",
        "vendor",
        ".turbo",
        ".pnpm-store",
        ".yarn",
        ".venv",
        "venv",
        "env",
        "__pycache__",
        ".pytest_cache",
        "site-packages",
        ".cursor",
        ".mypy_cache",
        ".ruff_cache",
        ".tox",
        ".cache",
    }
)

IGNORE_RELATIVE_PREFIXES = (".swarm/cache", ".swarm/reports")


def normalize_relative_path(path: str) -> str:
    return path.replace("\\", "/").strip()


def should_ignore_scanned_path(rel_path: str) -> bool:
    normalized = normalize_relative_path(rel_path)
    if any(
        normalized == prefix or normalized.startswith(f"{prefix}/")
        for prefix in IGNORE_RELATIVE_PREFIXES
    ):
        return True
    for segment in normalized.split("/"):
        if segment in IGNORE_DIR_NAMES:
            return True
    return False
